### PR TITLE
Switch to slog

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: Ben Ye <benye@amazon.com>

Update Prometheus version to https://github.com/prometheus/prometheus/pull/14906

### CODE REVIEW

**Comment by yeya24** [comment]
@harry671003 Can you help review this PR? This is a small change.
I checked the test case failure are all related to `round` function

**Comment by harry671003** [comment]
Approving. Thanks Ben.

**Review by harry671003 — APPROVED** [review]

File: engine/engine_test.go [hunk 1]
```diff
@@ -48,11 +41,21 @@ import (
...
+	"github.com/thanos-io/promql-engine/execution/warnings"
+	"github.com/thanos-io/promql-engine/logicalplan"
+	"github.com/thanos-io/promql-engine/query"
+	"github.com/thanos-io/promql-engine/storage/prometheus"
 )
 
 func TestMain(m *testing.M) {
 	parser.EnableExperimentalFunctions = true
-	goleak.VerifyTestMain(m)
+	goleak.VerifyTestMain(m,
```
[the remark this answers is not in the document]
Comment (yeya24) [reply engine/engine_test.go:55]: I checked their master branch and the goroutine is still there https://github.com/census-instrumentation/opencensus-go/blob/master/stats/view/worker.go#L34.

Thanos main repo also ignores this function in the goleak test. But I am not sure why the problem only occur now. It should be always there

**Comment by yeya24** [comment]
Merged the PR and I will rename the branch to `update-prometheus-3.0`

**Pull request merged**